bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -504,8 +504,8 @@ public void added(Provider<Rule> provider, Rule element) {
             String rUID = ruleEngine.getUniqueId();
             ruleWithUID = initRuleId(rUID, element);
         }
-        super.added(provider, ruleWithUID);
```

Comment:
It is not available in registry, but it is already stored in RuleManagedProvider. IMHO added/updated events should be before the state events of rule. Otherwise we can  receive a state events for rules which are not available in the registry (when the add method is failed)